azure/scope/machinepool.go

```diff
@@ -405,8 +405,27 @@ func (m *MachinePoolScope) applyAzureMachinePoolMachines(ctx context.Context) er
 	}
 
 	existingMachinesByProviderID := make(map[string]infrav1exp.AzureMachinePoolMachine, len(ampms))
-	for _, machine := range ampms {
-		existingMachinesByProviderID[machine.Spec.ProviderID] = machine
+	for _, ampm := range ampms {
+		// propagate Machine delete annotation from owner machine to AzureMachinePoolMachine
+		// this ensures setting a deleteMachine annotation on the Machine has an effect on the AzureMachinePoolMachine
+		// and the deployment strategy.
+		machine, err := util.GetOwnerMachine(ctx, m.client, ampm.ObjectMeta)
+		if err != nil {
+			// TODO(mw): which verbosity? info or error?
+			log.V(4).Info("failed to get owner machine", "machine", ampm.Spec.ProviderID)
+			continue
+		}
+		if machine != nil && machine.Annotations != nil {
+			if _, hasDeleteAnnotation := machine.Annotations[clusterv1.DeleteMachineAnnotation]; hasDeleteAnnotation {
+				log.V(4).Info("propagating DeleteMachineAnnotation", "machine", ampm.Spec.ProviderID)
+				if ampm.Annotations == nil {
+					ampm.Annotations = make(map[string]string)
+				}
+				ampm.Annotations[clusterv1.DeleteMachineAnnotation] = machine.Annotations[clusterv1.DeleteMachineAnnotation]
```

Comment:
Just to be clear, this does not update the AMPM in Kubernetes to have the delete annotation, right?

Reply:
I'll verify this tomorrow - I'm pretty sure not, unless there's a deferred function updating the AMPM.

Reply:
Yeah, so if we're not actually updating the AMPM labels in K8s and just writing it to the local var labels, we should change the title since "propagate delete Machine annotation" seems to imply that it would be written to the actual K8s objects.